ckanext/ogdchcommands/logic.py

```diff
@@ -235,3 +236,36 @@ def ogdch_shacl_validate(context, data_dict):  # noqa
                     shaclparser.resultdictkey_parseerror: e
                 }
                 writer.writerow(resultdict)
+
+
+def ogdch_cleanup_resources(context, data_dict):
+    """
+    cleans up the database from resources that have been deleted
+    """
+    dryrun = data_dict.get('dryrun')
+    tk.check_access('resource_delete', context, data_dict)
+    delete_resources = model.Session.query(model.Resource) \
+        .filter(model.Resource.state == 'deleted') \
+        .all()
+    delete_resources_ids = [resource.id for resource in delete_resources]
+    count = len(delete_resources_ids)
+
+    sql = '''begin;
```

Comment:
Is it neccessary to delete `resource_view` and `resource_revision` via SQL? Is there no action for that? 
If yes, I would highly prefer it that way to be less intrusive. If not, it might make sense to provide the deletion logic to CKAN-core, too.

Reply:
All our cleanup commands so far go directly to the database. Same for harvest_job_cleanup. There is an action resource_delete in ckan, but it only sets the resource to the state 'deleted'. So yes I think it is needed.

Reply:
thanks for double-checking and clarifying :)